### PR TITLE
refactor:eventhub/_eventprocessor:type hint added

### PR DESCRIPTION
**Pull request by JJHWAN**
# Description

issue #25471

Changed \eventhub\azure-eventhub\azure\eventhub\_eventprocessor directory with type hints

Will work with other dir too, later :)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py [hunk 1]
```diff
@@ -48,12 +49,11 @@ class EventProcessor(
...
     def __init__(
         self,
-        eventhub_client,  # type: EventHubConsumerClient
-        consumer_group,  # type: str
-        on_event,  # type: Callable[[PartitionContext, Union[Optional[EventData], List[EventData]]], None]
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> None
+        eventhub_client: EventHubConsumerClient,
+        consumer_group: str,
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py:53]: Looks like there are trailing whitespaces in many places. You'll need to remove these to pass CI. Ex.
```suggestion
        consumer_group: str,
```
[[Here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1849973&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=56d4e2e6-c43b-527c-bad7-234ebe7429dd&l=461)] is the list of all the places that have trailing whitespace.

File: sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/in_memory_checkpoint_store.py [hunk 2]
```diff
@@ -143,8 +149,7 @@ def claim_ownership(self, ownership_list, **kwargs):
                 result.append(ownership)
         return result
 
-    def update_checkpoint(self, checkpoint, **kwargs):
-        # type: (Dict[str, Optional[Union[str, int]]], Any) -> None
+    def update_checkpoint(self, checkpoint: str, **kwargs: Optional[Union[str, int]]) -> None:
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/in_memory_checkpoint_store.py:152]: I believe this should be the following:
```suggestion
    def update_checkpoint(self, checkpoint: Dict[str, Optional[Union[str, int]]], **kwargs: Any) -> None:
```

**Review by swathipil — COMMENTED** [review]
Hi @JJHWAN - Thanks for working on updating these type hints! I've left a few comments and can take another look once those have been addressed.

**Comment by JJHWAN** [comment]
@swathipil  applied your reviews!

**Review by swathipil — APPROVED** [review]

**Pull request merged**